src/.nuget/Microsoft.NET.Sdk.Optimization.ReadyToRun/sdk/Sdk.targets

```diff
@@ -0,0 +1,213 @@
+<!--
+***********************************************************************************************
+Sdk.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved.
+***********************************************************************************************
+-->
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!-- Remaining Work:
+    1) Inject CreateReadyToRunImages at correct entrypoint in the build. Related to https://github.com/dotnet/sdk/issues/2583
+  -->
+  <PropertyGroup>
+    <_IntermediateReadyToRunDirName Condition=" '$(_IntermediateReadyToRunDirName)' == '' ">ReadyToRun</_IntermediateReadyToRunDirName>
+    <_IntermediateReadyToRunDir Condition=" '$(_IntermediateReadyToRunDir)' == '' ">$(IntermediateOutputPath)$(_IntermediateReadyToRunDirName)</_IntermediateReadyToRunDir>
+    <_IntermediateReadyToRunDir>$([System.IO.Path]::GetFullPath($(_IntermediateReadyToRunDir)))</_IntermediateReadyToRunDir>
+
+    <_ReadyToRunOutputDir>$([System.IO.Path]::Combine($(_IntermediateReadyToRunDir),"Output"))</_ReadyToRunOutputDir>
+  </PropertyGroup>
+
+  <Target Name="CreateReadyToRunImages"
+          BeforeTargets="ComputeFilesToPublish"
+          DependsOnTargets="_CheckForUnsupportedReadyToRunCompilation;
+                            _InitializeReadyToRunCompilerProps;
+                            _ComputeReadyToRunInputsAndOutputs;
+                            _RunReadyToRunCompiler;
+                            _PublishReadyToRunOutputs"
+          Condition="'$(ReadyToRun)' == 'true'">
+  </Target>
+
+  <Target Name="_CheckForUnsupportedReadyToRunCompilation"
+          Condition="'$(ReadyToRun)' == 'true'">
+
+    <!-- TODO: Add this message as a resource string to the NET SDK so it can be localized -->
+    <NETSdkError Condition="'3.0' > '$(_TargetFrameworkVersionWithoutV)'"
+        FormattedText="Creating ReadyToRun images is only supported for .NET Core 3.0 and above" />
+
+    <NETSdkError Condition="'$(RuntimeIdentifier)' ==''"
+                 ResourceName="RuntimeIdentifierWasNotSpecified"/>
+
+    <NETSdkError Condition="'$(TargetFramework)' ==''"
+                 ResourceName="AtLeastOneTargetFrameworkMustBeSpecified"/>
```

Comment:
I am not comfortable with this coupling. NETSdkError is not meant to be used outside of dotnet/sdk. This will break our ability to look for unused resources, etc.